RELEASE_NOTES.md

```diff
@@ -6,6 +6,7 @@
 ## Other Changes
 * Optional permissions for `PersistentVolumeClaim` moved to a separate role. When managing the operator with Helm it is possible to disable permissions for `PersistentVolumeClaim` resources by setting `operator.enablePVCResize` value to `false` (`true` by default). When enabled, previously these permissions were part of the primary operator role. With this change, permissions have a separate role.
 * `subresourceEnabled` Helm value was removed. This setting used to be `true` by default and made it possible to exclude subresource permissions from the operator role by specifying `false` as the value. We are removing this configuration option, making the operator roles always have subresource permissions. This setting was introduced as a temporary solution for [this](https://bugzilla.redhat.com/show_bug.cgi?id=1803171) OpenShift issue. The issue has since been resolved and the setting is no longer needed.
+* The subcommand `debug` of `kubectl mongodb` plugin has been removed in favour of `mdb_operator_diagnostic_data.sh`. It's recommended to use the script `mdb_operator_diagnostic_data.sh` to gather data about MongoDB Controllers for Kubernetes and its resources.
```

Comment:
I don't think any release notes are required. We've never actually released the tool publicly in any release notes nor internally, so we're safe to remove it without any notice.

Reply:
It was added in response to [my review](https://github.com/mongodb/mongodb-kubernetes/pull/293#pullrequestreview-3051225185).

My logic was - it is already part of CLI and anyone can see it via `--help`. I don't know if anyone actually uses it or not.

I don't have any objections making this change without a release note.

But if we do add a release note - would be great to add one with the new tooling so we have less syncing to do once we switch over (see `changelog` dir in master and https://github.com/mongodb/mongodb-kubernetes/pull/193).